### PR TITLE
Merge code generation changes to main

### PR DESCRIPTION
**Pull request by lmolkova**
Brings all code generation changes to the main branch

### CODE REVIEW

**Review by jsuereth — APPROVED** [review]

**Review by trisch-me — APPROVED** [review]
lgtm assuming these are commits from already approved and merged PRs

File: semantic-conventions/README.md [hunk 1]
```diff
@@ -104,55 +136,332 @@ By default, all models are fed into the specified template at once, i.e. only a
...
 via the additional flag `--trim-whitespace`. Providing the flag will enable both `lstrip_blocks` and `trim_blocks`.
 
 ### Enabling/disabling support for colored diffs in error messages
 The `COLORED_DIFF` environment variable is set in the `semantic-conventions` `Dockerfile`.  When this environment varibale is set, errors related to reformatting tables will show a "colored diff" using standard ANSI control characters. While this should be supported natively in any modern terminal environment, you may unset this variable if issues arise.  Doing so will enable a "fall back" of non-colored inline diffs showing what was "added" and what was "removed", followed by the exact tokens added/removed encased in single quotes.
 
-## Version compatibility check
+### Accessing Semantic Conventions in the template
 
-You can check compatibility between the local one specified with `--yaml-root` and sepcific OpenTelemetry semantic convention version using the following command:
+When template is processed, it has access to a set of variables that depends on the `--file-per-group` value (or lack of it).
```
Comment (joaopgrassi) [inline semantic-conventions/README.md:196]:
```suggestion
When the template is processed, it has access to a set of variables that depends on the `--file-per-group` value (or lack of it).
```

File: semantic-conventions/README.md [hunk 2]
```diff
@@ -104,55 +136,332 @@ By default, all models are fed into the specified template at once, i.e. only a
...
+
+Processes a single pattern value and is called for each distinct value.
+
+- `semconv` - the instance of parsed `BaseSemanticConvention` being processed.
+
+### Filtering and mapping
+
+Jinja templates has a notion of [filters](https://jinja.palletsprojects.com/en/2.11.x/templates/#list-of-builtin-filters) allowing to transform objects or filter lists.
+
+Semconvgen supports following additional filters to simplify common operations in templates.
```
Comment (joaopgrassi) [inline semantic-conventions/README.md:229]:
```suggestion
Semconvgen supports the following additional filters to simplify common operations in templates.
```

File: semantic-conventions/README.md [hunk 3]
```diff
@@ -104,55 +136,332 @@ By default, all models are fed into the specified template at once, i.e. only a
...
+   The first character of every word is upper-cased, other characters are lower-cased. E.g. `foo.bAR_baz` becomes `fooBarBaz`
+4. `to_const_name` - Converts a string to Python or Java constant name (SNAKE_CASE) replacing `.` or `-` with `_`. E.g.
+   `foo.bAR-baz` becomes `FOO_BAR_BAZ`.
+5. `to_doc_brief` - Trims whitespace and removes dot at the end. E.g. ` Hello world.\t` becomes `Hello world`
+
+#### `BaseSemanticConvention` operations
+
+1. `is_metric` - Checks if semantic convention describes a metric.
+
+
```
Comment (joaopgrassi) [inline semantic-conventions/README.md:254]:
```suggestion
```

**Review by joaopgrassi — APPROVED** [review]

**Comment by joaopgrassi** [comment]
> lgtm assuming these are commits from already approved and merged PRs

If there's something not working or that needs ironing out, I'm sure we can do it soon after a release, so not much to risk here :)

**Comment by jack-berg** [comment]
🎉

**Pull request merged**